### PR TITLE
Added support for more platforms

### PR DESCRIPTION
**Pull request by llali**
- Added support for linuxmint and galliumos Linux distros ans win86
- Refactored the code in platform.ts 
- Trying /etc/os-release and fallback to /usr/lib/os-release to get linux os release info
- setting a display name for each runtime that is used for the folder name when the service is installed. The reason the display name is different than the runtime is that if the service is already installed and user gets the update comes with this PR, the extension won't try to install the service again just because the folder name is different

### CODE REVIEW

**Comment by llali** [comment]
#364

**Comment by llali** [comment]
Fix for #364

File: src/configurations/dev.config.json [hunk 1]
```diff
@@ -3,15 +3,16 @@
...
-            "OSX": "osx-x64-netcoreapp1.0.tar.gz",
-            "CentOS": "centos-x64-netcoreapp1.0.tar.gz",
-            "Debian": "debian-x64-netcoreapp1.0.tar.gz",
-            "Fedora": "fedora-x64-netcoreapp1.0.tar.gz",
-            "OpenSUSE": "opensuse-x64-netcoreapp1.0.tar.gz",
-            "RHEL": "rhel-x64-netcoreapp1.0.tar.gz",
-            "Ubuntu14": "ubuntu14-x64-netcoreapp1.0.tar.gz",
-            "Ubuntu16": "ubuntu16-x64-netcoreapp1.0.tar.gz"
+            "Windows_7_64": "win-x64-netcoreapp1.0.zip",
+            "Windows_7_86": "win-x86-netcoreapp1.0.zip",
```
Comment (kevcunnane) [inline src/configurations/dev.config.json]: We may not want to support this. C# doesn't and we're moving away from this in general. Given the very small number of users, I think we should take this out.

**Comment by kevcunnane** [comment]
:shipit:

**Pull request merged**